Added a new algorithm to the OIT sample

mdagois opened the pull request:
* Added the first buffer-based algorithm (buckets)
* Added a way to find out if the hardware supports fragment shader stores and atomics
* Fixed an issue that was preventing the creation of a draw pass without depth/stencil
* Performed some code clean up based on the previous PR comments

vettoreldaniele commented on projects/oit_demo/Buffer.cpp:
```diff
@@ -0,0 +1,294 @@
+        createInfo.height                     = mBuffer.countTexture->GetHeight();
+        createInfo.renderTargetCount          = 0;
+        createInfo.pDepthStencilImage         = nullptr;
+        createInfo.renderTargetClearValues[0] = {0, 0, 0, 0};
```
Since we have no render targets, you can omit this

vettoreldaniele commented on projects/oit_demo/Buffer.cpp:
```diff
@@ -0,0 +1,294 @@
+
+        PPX_CHECKED_CALL(GetDevice()->AllocateDescriptorSet(mDescriptorPool, mBuffer.gatherDescriptorSetLayout, &mBuffer.gatherDescriptorSet));
+
+        grfx::WriteDescriptor writes[4] = {};
```
Use `std::array<WriteDescriptor, 4>` so you can do `.size()` later.  Same for similar uses below

mdagois replied:
I switched to using std::array for WriteDescriptors in all algorithms.

vettoreldaniele commented on projects/oit_demo/Buffer.cpp:
```diff
@@ -0,0 +1,294 @@
+
+void OITDemoApp::RecordBuffer()
+{
+    static bool sTextureNeedClear = true;
```
Can you move this inside of `mBuffer` to avoid using static?

mdagois replied:
I moved the variable into mBuffer. Fixed.

vettoreldaniele commented on projects/oit_demo/Buffer.cpp:
```diff
@@ -0,0 +1,294 @@
+    }
+
+    {
+        mCommandBuffer->TransitionImageLayout(mBuffer.countTexture, 0, 1, 0, 1, grfx::RESOURCE_STATE_SHADER_RESOURCE, grfx::RESOURCE_STATE_GENERAL);
```
We generally use `PPX_ALL_SUBRESOURCES`, e.g.:
```
mCommandBuffer->TransitionImageLayout(mBuffer.countTexture, PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_SHADER_RESOURCE, grfx::RESOURCE_STATE_GENERAL);
```

vettoreldaniele commented:
Seems there is a conflicting file (probably needs rebase?)

mdagois commented:
> Seems there is a conflicting file (probably needs rebase?)

Someone had fixed a warning I had also fixed. I rebased to the latest main. That should do the trick.

vettoreldaniele commented:
I think the README update is missing, otherwise looks fine to me

mdagois commented:
> I think the README update is missing, otherwise looks fine to me

Oh my! You are right. I forgpt the reafme. I will update it tomorrow and push again. :)

mdagois commented:
I updated the README. Please have a look.
